server/src/main/java/org/opensearch/search/aggregations/bucket/histogram/InternalDateHistogram.java

```diff
@@ -394,9 +395,47 @@ protected Bucket reduceBucket(List<Bucket> buckets, ReduceContext context) {
         return createBucket(buckets.get(0).key, docCount, aggs);
     }
 
-    private void addEmptyBuckets(List<Bucket> list, ReduceContext reduceContext) {
+    private int getTotalBucketCount() {
+        LongBounds bounds = emptyBucketInfo.bounds;
```

Comment:
If the bounds here is the extended bounds provided by user query, do we also cover the case that user doesn't give extended bounds, but the date values are separated a lot, like there's only 1970-01-01, and 3000-01-01, and there could be many empty buckets added I guess (not sure if this is true, I remeber for histogram that's the case).

Reply:
I provide a simple case below.

We cannot just rely on emptyBucket to estimate the total bucket count
We should also consider the min and max from the reducedBuckets, which is calculated before add empty bucket

The logic would be

min = math.min(emptyBucketInfo.min, reducedBuckets.minBucket.bucketKeyvalue)
max = math.max(emptyBucketInfo.max, reducedBuckets.maxBucket.bucketKeyvalue)

estimateTotalBucket = (max-min)/interval